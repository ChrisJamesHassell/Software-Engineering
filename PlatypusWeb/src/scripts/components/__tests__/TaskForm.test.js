import React from 'react';
import { shallow, mount, render } from 'enzyme';
import TaskForm from '../Forms/TaskForm';

const props = { 
    category: null,
    deadline: '',
    description: '',
    name: '',
    priority: null,
  };

describe('TaskForm', () =>{
    const taskForm = shallow(<TaskForm  />  );


    it('should render Login Container', () => {
        expect(taskForm.find('Formik').exists()).toBe(true);
    });



describe('TaskForm', () =>{

    it('matches the snapshot', () =>{
        const tree = shallow(<TaskForm />)
        expect(tree).toMatchSnapshot()
        })
    })
})
