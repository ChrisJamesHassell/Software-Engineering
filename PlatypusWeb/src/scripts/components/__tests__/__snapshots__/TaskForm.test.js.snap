// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm TaskForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "enableReinitialize": false,
      "initialValues": Object {
        "category": null,
        "deadline": "",
        "description": "",
        "name": "",
        "priority": null,
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "enableReinitialize": false,
        "initialValues": Object {
          "category": null,
          "deadline": "",
          "description": "",
          "name": "",
          "priority": null,
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
